src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -1009,12 +1009,20 @@ export default class PlotsTab extends React.Component<IPlotsTabProps, {}> {
                     // select default if _selectedGenericAssayOption is undefined and there are generic assay options to choose from
                     // if there is a gene selected in the other axis, select the first related option in this axis
                     // this will not override the option recorded in the url
-                    const selectedGeneRelatedOptions = selectedHugoGeneSymbolInTheOtherAxis
-                        ? filterGenericAssayOptionsByGenes(
-                              genericAssayOptions,
-                              [selectedHugoGeneSymbolInTheOtherAxis]
-                          )
-                        : [];
+                    const genericAssayType = vertical
```

Comment:
this logic is pretty difficult to read.  we should improve the commenting to explain why this is necessary.  
comments line by line.